### ISSUE TITLE
Regarding the reproduction test

### ISSUE THREAD

**Issue by gonasoo93** [zhoubohan0/STG-Transformer#2]
I am interested in the paper's content and conducted a reproduction test. Still, when making a model (ex. sac_expert.py) for expert data generation or in the online reinforcement learning stage, the episodic return value returned by the model comes out differently from the paper or does not converge. Can you tell me more about the experimental conditions or check if there are any problems with the code you uploaded?

**Comment by zhoubohan0** [comment]
Thanks for your interest. We provide SAC for pretraining expert agent for data generation and PPO for STG base algorithm in the repository. The pretrained SAC agent must converge in a certain task (**at least** 5M env steps) and is **only** responsible for providing a few near-optimal trajectories. It is meaningless to compare the episodic return values between them. We mainly focus on the gain that comes from offline pretraining to online task learning.

**Comment by gonasoo93** [comment]
Yes, I understand that part, too. I understand that the SAC algorithm is currently being used as a tool to provide expert data in your paper.

  I have not been able to obtain results similar to those of online learning results in the paper; therefore, I am analyzing various causes. So far, I have been experimenting with reproducing the Atari results. In offline learning, the model converged well and reached the online learning stage; however, the critical value loss was not stabilized and the episodic return did not increase online learning stage. Therefore, I am looking at the problems from the offline learning stage.

  Currently, most reproduction results have converged on the model in the offline learning stage, but not in the online stage. Can you give me some advice on what areas to look at to solve the problem?

**Comment by zhoubohan0** [comment]
Sorry for the delay. Maybe some of the hints below may help you reimplement the results:
1. During offline pretraining, we also noticed that the total loss declines smoothly and can easily get converged in early training stage but the downstream performance suffers. Ensure each loss item gets completely converged and make training sufficient. You can also visualize the embeddings to check.
2. In RL stage, we set different GAE \lambda for each task. This is important for satisfactory results.

**Comment by gonasoo93** [comment]
Thank you for letting me know!
